### PR TITLE
feat(templates): support same-repo templates using build commit

### PR DESCRIPTION
**Pull request by ecrupper**
When working with templates that are found in the same repo as the Vela file, any live edits to the templates requires the users to reference their specific branch in the templates block. Example: 
```yaml
templates:
  - name: sample
    source: https://github.com/<org>/<repo>/path/template.yml@feature-branch
    type: github
```

With this change, users can specify the type `file` in their templates block, which will automatically fetch the template from the commit associated with the build. Example: 
```yaml
templates:
  - name: sample
    source: path/template.yml
    type:  file
```

### CODE REVIEW

**Comment by JordanSussman** [comment]
@ecrupper I feel like we should have the `type` be either `github-commit` or `commit-github` incase we want to add additional source code systems in the provider.

File: compiler/native/expand_test.go [hunk 1]
```diff
@@ -287,25 +317,29 @@ func TestNative_ExpandSteps(t *testing.T) {
...
-	if err != nil {
-		t.Errorf("ExpandSteps returned err: %v", err)
-	}
+	compiler.WithBuild(testBuild).WithRepo(testRepo)
 
-	if diff := cmp.Diff(build.Steps, wantSteps); diff != "" {
-		t.Errorf("ExpandSteps() mismatch (-want +got):\n%s", diff)
-	}
+	for _, test := range tests {
+		build, err := compiler.ExpandSteps(&yaml.Build{Steps: steps, Services: yaml.ServiceSlice{}, Environment: globalEnvironment}, test.tmpls)
```
Comment (cognifloyd) [inline compiler/native/expand_test.go:323]: Would you please use `t.Run()` here? That makes selecting subtests easier in Goland.

**Review by cognifloyd — CHANGES_REQUESTED** [review]
Very nice!

I think we should stick with `commit` (or something similar) that does not mention github. See my comment.

I have one nitpick about the test. Otherwise, I say this is good to go!

**Comment by wass3r** [comment]
this would be really nice for local pipeline execution via `vela exec pipeline` - even without checking code in. but i think the current implementation doesn't suit itself for that. part of the reason i thought this can be scm agnostic and the type be `relative` or `file`. thoughts around that?

**Comment by cognifloyd** [comment]
> this would be really nice for local pipeline execution via `vela exec pipeline` - even without checking code in. but i think the current implementation doesn't suit itself for that. part of the reason i thought this can be scm agnostic and the type be `relative` or `file`. thoughts around that?

Is that what `c.local` bit does?

https://github.com/go-vela/server/blob/56932446fdc545d44162da499f7318ad34994c77/compiler/native/expand.go#L170-L179

I kind of like `relative` - but that might suggest that it is relative to the current file vs relative to the repo root.
I think I could grow to like `file` even better. The more I think about this one, I think I like it a lot.

**Comment by wass3r** [comment]
> Is that what c.local bit does?

busted! forgot about that. 🫣

**Review by wass3r — APPROVED** [review]

**Comment by wass3r** [comment]
@cognifloyd i think your feedback is addressed

@ecrupper we will need to add to docs too

**Review by cognifloyd — APPROVED** [review]

**Pull request merged**